iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -253,13 +243,58 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     void destruct_value() noexcept;
 };
 
+// AXIVION Next Construct AutosarC++19_03-M17.0.3 : make_optional is defined within iox::cxx which prevents easy misuse
 /// @brief Creates an optional which contains a value by forwarding the arguments
 ///         to the constructor of T.
 /// @param[in] args arguments which will be perfectly forwarded to the constructor
 ///             of T
 /// @return optional which contains T constructed with args
 template <typename OptionalBaseType, typename... Targs>
 optional<OptionalBaseType> make_optional(Targs&&... args) noexcept;
+
+/// @brief Compare two optionals for equality.
+/// @param[in] lhs cxx::optional
+/// @param[in] rhs value to which this optional should be compared to
+/// @return true if the contained values are equal or both have no value, otherwise false
+template <typename T>
+bool operator==(const optional<T>& lhs, const optional<T>& rhs) noexcept;
+
+/// @brief Compare two optionals for inequality.
+/// @param[in] lhs cxx::optional
+/// @param[in] rhs value to which this optional should be compared to
+/// @return true if the contained values are not equal, otherwise false
+template <typename T>
+bool operator!=(const optional<T>& lhs, const optional<T>& rhs) noexcept;
+
+// AXIVION DISABLE STYLE AutosarC++19_03-A13.5.5: Comparison with nullopt_t is required
+/// @brief Comparison for equality with nullopt_t for easier unset optional comparison
+/// @param[in] lhs empty optional, cxx::nullopt_t
+/// @param[in] rhs value to which this optional should be compared to
+/// @return true if the optional is set, otherwise false
+template <typename T>
+bool operator==(const nullopt_t, const optional<T>& rhs) noexcept;
+
+/// @brief Comparison for equality with nullopt_t for easier unset optional comparison
+/// @param[in] lhs value to which this optional should be compared to
+/// @param[in] rhs empty optional, cxx::nullopt_t
+/// @return true if the optional is set, otherwise false
+template <typename T>
+bool operator==(const optional<T>& lhs, const nullopt_t) noexcept;
+
+/// @brief Comparison for inequality with nullopt_t for easier unset optional comparison
+/// @param[in] lhs empty optional, cxx::nullopt_t
+/// @param[in] rhs value to which this optional should be compared to
+/// @return true if the optional is set, otherwise false
+template <typename T>
+bool operator!=(const nullopt_t, const optional<T>& rhs) noexcept;
```

Comment:
Needs to be tested in `test_cxx_optional.cpp`.

Reply:
@MatthiasKillat New tests are added.